raft/fsm.hh

```diff
@@ -28,6 +28,7 @@ struct fsm_output {
     // Entries to apply.
     std::vector<log_entry_ptr> committed;
     std::optional<applied_snapshot> snp;
+    std::vector<snapshot_id> snps_to_drop;
```

Comment:
We may re-allocate the vector here when we add elements. All other vectors used in the output are pre-allocated to the correct size. We should either use list here or may be `utils::small_vector` and hope we never allocate.